app/assets/javascripts/discourse/app/components/modal/edit-user-directory-columns.js

```diff
@@ -0,0 +1,106 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
+import { ajax } from "discourse/lib/ajax";
+import { extractError, popupAjaxError } from "discourse/lib/ajax-error";
+import { reload } from "discourse/helpers/page-reloader";
+
+const UP = "up";
+const DOWN = "down";
+
+export default class EditUserDirectoryColumns extends Component {
+  @tracked loading = true;
+  @tracked columns;
+  @tracked labelKey;
+  @tracked flash;
+
+  constructor(owner, args) {
+    super(owner, args);
```

Comment:
Just for consistency:

```suggestion
  constructor() {
    super(...arguments);
```